test/unit/test_cons.c

```diff
@@ -324,13 +324,48 @@ bool test_line_kill_word(void) {
 	mu_end;
 }
 
+bool test_line_undo(void) {
+	RzCons *cons = rz_cons_new();
+	// Make test reproducible everywhere
+	cons->force_columns = 80;
+	cons->force_rows = 23;
+	rz_line_free();
+	RzLine *line = rz_line_new();
+	line->ns_completion.run = onecompletion_run;
+	cons->line = line;
+
+	// write 20 chars and undo once
```

Comment:
is the "undo" the `\x1f`? Is the `\n` after `\x1f` required?

Reply:
undo is '\x1f' and redo is '\x1b\x3f'. For '\n', it was just needed for return from readline func. I will do the tests in more direct way.

Reply:
It appears that I have to use '\n'.